alpha/apps/kaltura/lib/clipconcat/kClipManager.php

```diff
@@ -0,0 +1,509 @@
+				break;
+
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * @param BatchJob $parentJob  clipConcat job
+	 * @param $entryId
+	 * @param $errDescription
+	 * @param $partnerId
+	 * @param array $operationAttributes
+	 * @param int $priority
+	 * @return BatchJob[]
+	 * @throws APIException
+	 * @throws PropelException
+	 */
+	private function addClipJobs($parentJob , $entryId, &$errDescription, $partnerId,
+	                             array $operationAttributes, $priority = 0)
+	{
+		$batchArray = array();
+		$order = 0;
+		foreach($operationAttributes as $singleAttribute)
+		{
+			KalturaLog::info("Going To create Flavor for clip: " . print_r($singleAttribute));
+			$clonedID =	$this->cloneFlavorParam($singleAttribute->getAssetParamsId());
+			$flavorAsst = $this->createTempClipFlavorAsset($partnerId,$entryId,$clonedID,$order);
+			$batchJob =	kBusinessPreConvertDL::decideAddEntryFlavor($parentJob, $entryId,
+					$clonedID, $errDescription,$flavorAsst->getId()
+					, array($singleAttribute) , $priority);
+			if(!$batchJob) //todo error;
+					return null;
+			$batchArray[] = $batchJob;
+			$order++;
+		}
+		return $batchArray;
+	}
+
+
+	/***
+	 * @param BatchJob $batchJob
+	 * @return bool are all clip batch done
+	 */
+	private function handleClipChildJob($batchJob)
+	{
+
+		$childJobs = $batchJob->getRootJob()->getChildJobs();
+		/**
+		 * check if all child jobs are finished(all clip jobs and then a single concat job)
+		 * @var BatchJob $job */
+		foreach ($childJobs as $job)
+		{
+			if ($job->getStatus() != BatchJob::BATCHJOB_STATUS_FINISHED )
+			{
+				KalturaLog::info("number of children:   ". count($childJobs));
+				KalturaLog::info('Child job id [' . $job->getId() . '] status [' . $job->getStatus() . ']');
+				return false;
+			}
+		}
+		if ($batchJob->getJobType() == BatchJobType::POSTCONVERT  &&
+			$batchJob->getStatus() == BatchJob::BATCHJOB_STATUS_FINISHED)
+		{
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * @param $assets
+	 * @return array
+	 * @throws kCoreException
+	 * @throws Exception
+	 */
+	private function getFilesPath($assets): array
+	{
+		$files = array();
+		foreach ($assets as $asset) {
+			/**
+			 * Don't take source it is empty
+			 * @var flavorAsset $asset */
+			if ($asset->getIsOriginal()) {
+				continue;
+			}
+			$syncKey = $asset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET);
+			$fileSync = kFileSyncUtils::getReadyFileSyncForKey($syncKey);
+			//fileSync should be local
+			if (!$fileSync[1])
+			{
+				throw new kCoreException("Clip Does not exist locally operations stops, there will be no concat");
+			}
+			/*** @var array $fileSync */
+			if ($fileSync[0]->getFullPath())
+			{
+				$files[] = $fileSync[0]->getFullPath();
+			}
+		}
+		return $files;
+	}
+
+	/**
+	 * @param entry $tempEntry
+	 * @return flavorAsset
+	 * @throws kCoreException
+	 */
+	private function getNewAssetFromEntry($tempEntry): flavorAsset
```

Comment:
this function adds new asset to entry , its name is confusing , especially when get.. functions must not have side effect.